site/docs/design-docs/018-kinesis-connectors.md

```diff
@@ -0,0 +1,482 @@
+  where we need the most control possible to make choices that are
+  suitable to Jet's architecture.
+* **Amazon Kinesis Data Streams API** via **AWS SDK for Java** is the
+  lower-level library that allows sufficient control when interacting
+  with Kinesis. It consists of a simple set of [REST-based
+  operations](https://docs.aws.amazon.com/kinesis/latest/APIReference/API_Operations.html).
+  Every other concern mentioned above, when discussing the high-level
+  libraries, has to be handled explicitly. This is the library used in
+  the Jet source and sink implementations.
+
+## Quotas
+
+Amazon Kinesis Data Streams enforces quite a few quotas and limits,
+which our sources and sinks need to comply with:
+
+* A single shard can ingest up to 1 MiB of data per second (including
+  partition keys) or 1,000 records per second for writes.
+* The maximum size of the data payload of a record is 1 MiB.
+* The
+  [GetRecords](https://docs.aws.amazon.com/kinesis/latest/APIReference/API_GetRecords.html)
+  operation can retrieve up to 10 MiB of data per call from a single
+  shard and up to 10,000 records per call.
+* Each shard can support up to 5 GetRecords operations per second.
+* Each shard can support up to a maximum total data read rate of 2 MiB
+  per second.
+* The
+  [ListShards](https://docs.aws.amazon.com/kinesis/latest/APIReference/API_ListShards.html)
+  operation has a limit of 100 transactions per second, per data stream.
+  Each such transaction is able to return at most 100 shards, and if the
+  stream has more, then multiple transactions need to be used for a full
+  listing. (For details, see the [discovery](#discovery) section.)
+* The
+  [PutRecords](https://docs.aws.amazon.com/kinesis/latest/APIReference/API_PutRecords.html)
+  operation can write at most 500 records into the stream. Each record
+  in the request can be as large as 1MiB, up to a limit of 5MiB for the
+  entire request, including partition keys. Each shard can support
+  writes up to 1,000 records per second, up to a maximum data write
+  total of 1 MiB per second.
+* The
+  [DescribeStreamSummary](https://docs.aws.amazon.com/kinesis/latest/APIReference/API_DescribeStreamSummary.html)
+  operation has a limit of 20 transactions per second per account. (For
+  details, see the [adaptive throughput](#adaptive-throughput) section.)
+
+## Source
+
+The Kinesis source is a _streaming_, _distributed_, and _fault-tolerant_
+data source for Jet. It supports both the _at-least-once_ and
+_exactly-once_ [processing
+guarantees](../architecture/fault-tolerance.md#processing-guarantee-is-a-shared-concern).
+
+### Distribution
+
+Being a distributed source, it has multiple instances running in each
+Jet cluster member. Each instance is responsible for reading from zero,
+one or more KDS [shards](#shard). The assignment of shards to individual
+source instances is based on partitioning the space of Kinesis [record
+keys](#partition-key).
```

Comment:
Asssignment of processors isn't based on "record keys". It seems as if one shard could be read by multiple processors based on the record key.